src/ml.jl

```diff
@@ -0,0 +1,75 @@
+# Not correct for `map(pick(:foo), t)`
+Base.names(t::DTable) = fieldnames(eltype(t))
+Base.names(t::IndexedTable) = fieldnames(eltype(t))
+
+# Core schema types
+
+width(::Void) = 0
+numeric(::Void, x) = []
+
+struct Continuous{T}
+  μ::T
+  σ::T
+end
+
+width(::Continuous) = 1
+numeric(c::Continuous, x) = (x - c.μ) / c.σ
+
+struct Categorical{T}
+  items::Vector{T}
+end
+
+width(c::Categorical) = length(c.items)
```

Comment:
If a model is linear in the parameters, you can represent `k` levels with `k-1` indicator variables.  It would be great if we could dispatch on the encoding.  Apparently StatsModels does this: https://juliastats.github.io/StatsModels.jl/latest/contrasts.html